### PR TITLE
Requests validations for chapter 1

### PR DESCRIPTION
**Pull request by kamilbaczek**

### CODE REVIEW

File: Chapter-1-initial-architecture/Src/Fitnet.UnitTests/Contracts/PrepareContract/PrepareContractRequestValidatorTests.cs [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        
+        // Act
+        var result = _validator.TestValidate(request);
+
+        // Assert
+        result.ShouldHaveValidationErrorFor(prepareContractRequest => prepareContractRequest.CustomerAge);
+    }
+
+    [Fact]
+    public void Given_PrepareContractRequest_With_Invalid_CustomerHeight_Should_Have_Validation_Error()
```
Comment (kamilbaczek) [inline Chapter-1-initial-architecture/Src/Fitnet.UnitTests/Contracts/PrepareContract/PrepareContractRequestValidatorTests.cs:40]: Adjust names to our tests names convention

File: Chapter-1-initial-architecture/Src/Fitnet.UnitTests/Contracts/PrepareContract/PrepareContractRequestValidatorTests.cs [hunk 2]
```diff
@@ -0,0 +1,66 @@
+namespace EvolutionaryArchitecture.Fitnet.UnitTests.Contracts.PrepareContract;
```
Comment (kamilbaczek) [inline Chapter-1-initial-architecture/Src/Fitnet.UnitTests/Contracts/PrepareContract/PrepareContractRequestValidatorTests.cs:1]: Add to docs fluent validation and structor

File: Chapter-1-initial-architecture/Src/Fitnet/Common/Validation/Requests/RequestValidationsExtensions.cs [hunk 3]
```diff
@@ -0,0 +1,20 @@
+namespace EvolutionaryArchitecture.Fitnet.Common.Validation.Requests;
+
+using FluentValidation;
+using Logging;
+
+internal static class RequestValidationsExtensions
+{
+    internal static IServiceCollection AddRequestsValidations(this IServiceCollection services)
```
Comment (meaboutsoftware) [inline Chapter-1-initial-architecture/Src/Fitnet/Common/Validation/Requests/RequestValidationsExtensions.cs:8]: You do not need this extension, can use:

builder.Services.AddValidatorsFromAssemblyContaining<Program>();

Consider it :)

File: Chapter-1-initial-architecture/Src/Fitnet/Common/Validation/Requests/Logging/RequestValidationLogger.cs [hunk 4]
```diff
@@ -0,0 +1,22 @@
+namespace EvolutionaryArchitecture.Fitnet.Common.Validation.Requests.Logging;
+
+using System.Text;
+
+internal sealed class RequestValidationLogger : IRequestValidationLogger
```
Comment (meaboutsoftware) [inline Chapter-1-initial-architecture/Src/Fitnet/Common/Validation/Requests/Logging/RequestValidationLogger.cs:5]: I would omit logging completely (as I do not think we want to introduce Serilog in our repo at the moment).

File: Chapter-1-initial-architecture/Src/Fitnet.UnitTests/Contracts/SignContract/SignContractRequestValidatorTests.cs [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+
+using Fitnet.Contracts.SignContract;
+using FluentValidation.TestHelper;
+
+public sealed class SignContractRequestValidatorTests
+{
+    private readonly SignContractRequestValidator _validator = new();
+
+    [Fact]
+    public void Given_SignContractRequest_With_Invalid_SignedAt_Should_Have_Validation_Error()
```
Comment (meaboutsoftware) [inline Chapter-1-initial-architecture/Src/Fitnet.UnitTests/Contracts/SignContract/SignContractRequestValidatorTests.cs:11]: Usually when I write tests for FluentValidation I try to handle all cases with Theory. E.g. if I have validation for NotEmpty, I put a test case that validates:

Null,
Empty,
Whitespace

Something like here:

```
internal static class XyzRequestValidatorTestsData
{
    private static readonly Faker Faker = new();

    public static IEnumerable<object[]> GetIncorrectText()
    {
        yield return new object[] { string.Empty };
        yield return new object[] { " " };
        yield return new object[] { "<script>alert('xss')</script>" };
        yield return new object[] { Faker.Random.String2(5001) };
    }
}
```

Then you have the usage as follows:

```
[Theory]
    [MemberData(nameof(XyztRequestValidatorTestsData.GetIncorrectText),
        MemberType = typeof(XyzRequestValidatorTestsData))]
```

**Review by meaboutsoftware — CHANGES_REQUESTED** [review]

**Comment by meaboutsoftware** [comment]
I really like the way how you solved resolver

File: Chapter-1-initial-architecture/Src/Fitnet/Common/Validation/Requests/RequestValidationsExtensions.cs (hunk 3 above)
Comment (kamilbaczek) [reply Chapter-1-initial-architecture/Src/Fitnet/Common/Validation/Requests/RequestValidationsExtensions.cs:8]: I  have looked for this extension method but I didn't found. I will use it

File: Chapter-1-initial-architecture/Src/Fitnet.UnitTests/Contracts/SignContract/SignContractRequestValidatorTests.cs (hunk 5 above)
Comment (kamilbaczek) [reply Chapter-1-initial-architecture/Src/Fitnet.UnitTests/Contracts/SignContract/SignContractRequestValidatorTests.cs:11]: I know this approach. I used this way to have explicit named test cases but I am okay to do it in the way like you suggested.

Comment (meaboutsoftware) [reply Chapter-1-initial-architecture/Src/Fitnet.UnitTests/Contracts/SignContract/SignContractRequestValidatorTests.cs:11]: You can do it with separate test cases but handle all of them, not only default ;)

**Review by meaboutsoftware — APPROVED** [review]

**Pull request merged**